panic when close ClientConn

### What version of gRPC are you using?
1.15.0

### What version of Go are you using (`go version`)?
1.11

### What operating system (Linux, Windows, …) and version?
linux

### What did you do?
Call Close on a ClientConn, meanwhile the resolver resolved a new service config

### What did you expect to see?
Close ClientConn without err

### What did you see instead?
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x910846]

goroutine 5513264 [running]:
myapp/vendor/google.golang.org/grpc.(*ccBalancerWrapper).handleResolvedAddrs(0x0, 0xc448770280, 0x1, 0x1, 0x0, 0x0)
        /opt/gopath/src/myapp/vendor/google.golang.org/grpc/balancer_conn_wrappers.go:182 +0x26
myapp/vendor/google.golang.org/grpc.(*ClientConn).handleServiceConfig(0xc43dc7ac00, 0xc445bfbc00, 0x1a7, 0x1, 0x1)
        /opt/gopath/src/myapp/vendor/google.golang.org/grpc/clientconn.go:1046 +0x299
myapp/vendor/google.golang.org/grpc.(*ccResolverWrapper).watcher(0xc43af18930)
        /opt/gopath/src/myapp/vendor/google.golang.org/grpc/resolver_conn_wrapper.go:125 +0x2d6
created by myapp/vendor/google.golang.org/grpc.(*ccResolverWrapper).start
        /opt/gopath/src/myapp/vendor/google.golang.org/grpc/resolver_conn_wrapper.go:95 +0x3f
```

--- a/clientconn.go
+++ b/clientconn.go
@@ -747,6 +747,13 @@ func (cc *ClientConn) handleServiceConfig(js string) error {
 		return err
 	}
 	cc.mu.Lock()
+	// Check if the ClientConn is already closed. Some fields (e.g.
+	// balancerWrapper) are set to nil when closing the ClientConn, and could
+	// cause nil pointer panic if we don't have this check.
+	if cc.conns == nil {
+		cc.mu.Unlock()
+		return nil
+	}
 	cc.scRaw = js
 	cc.sc = sc
 

--- a/clientconn_test.go
+++ b/clientconn_test.go
@@ -676,6 +676,22 @@ func TestResolverServiceConfigBeforeAddressNotPanic(t *testing.T) {
 	time.Sleep(time.Second) // Sleep to make sure the service config is handled by ClientConn.
 }
 
+func TestResolverServiceConfigWhileClosingNotPanic(t *testing.T) {
+	defer leakcheck.Check(t)
+	for i := 0; i < 10; i++ { // Run this multiple times to make sure it doesn't panic.
+		r, rcleanup := manual.GenerateAndRegisterManualResolver()
+		defer rcleanup()
+
+		cc, err := Dial(r.Scheme()+":///test.server", WithInsecure())
+		if err != nil {
+			t.Fatalf("failed to dial: %v", err)
+		}
+		// Send a new service config while closing the ClientConn.
+		go cc.Close()
+		go r.NewServiceConfig(`{"loadBalancingPolicy": "round_robin"}`) // This should not panic.
+	}
+}
+
 func TestResolverEmptyUpdateNotPanic(t *testing.T) {
 	defer leakcheck.Check(t)
 	r, rcleanup := manual.GenerateAndRegisterManualResolver()

